src/main/java/com/google/sps/Utility.java

```diff
@@ -207,6 +205,35 @@ public static MultiMutation getMultiMutationAtIndex(List<MultiMutation> multiMut
     return lst;
   }
 
+  /**
+   * Returns a set of indices on the original list that related to a given token
+   *
+   * @param tokenName the token name to search for
+   * @param origList the original list of mutations
+   * @return a set of indices, empty if tokenName is null or if token is not changed
+   */
+  public static Set<Integer> getMutationIndicesOfToken(
+      String tokenName, List<MultiMutation> origList) {
+    Set<Integer> lst = new HashSet<>();
+    if (tokenName == null || tokenName.length() == 0) {
+      return lst;
+    }
+    for (int i = 0; i < origList.size(); i++) {
+      MultiMutation multiMut = origList.get(i);
```

Comment:
nit: you don't need `multiMut` as a variable, so you can just omit it and say `mutList = origList.get(i).getMutationList();`